examples/benchmark/src/main.c

```diff
@@ -0,0 +1,150 @@
+		.name = (_name), .type = (_type), .params = (const int32_t[]){__VA_ARGS__},        \
+		.params_nb = ((size_t)(sizeof((const int32_t[]){__VA_ARGS__}) / sizeof(int32_t))), \
+		.fourcc = (_fourcc),                                                               \
+	}
+
+int benchmark_operation(enum mpix_op_type type, const int32_t *params, size_t params_nb,
+			uint32_t format)
+{
+	struct mpix_image img = {};
+	struct mpix_format fmt = {.width = WIDTH, .height = HEIGHT, .fourcc = format};
+	size_t img_size = mpix_format_pitch(&fmt) * fmt.height;
+	size_t outbuf_size = sizeof(dstbuf);
+	mpix_image_from_buf(&img, srcbuf, img_size, &fmt);
+	for (size_t i = 0; i < NUM_OPS_RUN; ++i) {
+		CHECK(mpix_pipeline_add(&img, type, params, params_nb));
+	}
+	CHECK(mpix_image_to_buf(&img, dstbuf, outbuf_size));
+	mpix_print_pipeline(img.first_op);
+	mpix_image_free(&img);
+	return 0;
+}
+
+int main(void)
+{
+	struct benchmark_desc bench_tests[] = {
+		/* correction */
+		BENCHMARK_DESC_AUTO("correct_white_balance", MPIX_OP_CORRECT_WHITE_BALANCE,
+				    MPIX_FMT_RGB24),
+		BENCHMARK_DESC_AUTO("correct_black_level", MPIX_OP_CORRECT_BLACK_LEVEL,
+				    MPIX_FMT_RGB24),
+		BENCHMARK_DESC_AUTO("correct_gamma", MPIX_OP_CORRECT_GAMMA, MPIX_FMT_RGB24),
+		BENCHMARK_DESC_AUTO("correct_combined_ccm", MPIX_OP_CORRECT_COLOR_MATRIX,
+				    MPIX_FMT_RGB24),
+
+		/* kernel */
+		BENCHMARK_DESC_AUTO("kernel_identity_3x3", MPIX_OP_KERNEL_CONVOLVE_3X3,
+				    MPIX_FMT_RGB24, MPIX_KERNEL_IDENTITY),
+		BENCHMARK_DESC_AUTO("kernel_identity_5x5", MPIX_OP_KERNEL_CONVOLVE_5X5,
+				    MPIX_FMT_RGB24, MPIX_KERNEL_IDENTITY),
+		BENCHMARK_DESC_AUTO("kernel_sharpen_3x3", MPIX_OP_KERNEL_CONVOLVE_3X3,
+				    MPIX_FMT_RGB24, MPIX_KERNEL_SHARPEN),
+		BENCHMARK_DESC_AUTO("kernel_sharpen_5x5", MPIX_OP_KERNEL_CONVOLVE_5X5,
+				    MPIX_FMT_RGB24, MPIX_KERNEL_SHARPEN),
+		BENCHMARK_DESC_AUTO("kernel_edgedetect_3x3", MPIX_OP_KERNEL_CONVOLVE_3X3,
+				    MPIX_FMT_RGB24, MPIX_KERNEL_EDGE_DETECT),
+		BENCHMARK_DESC_AUTO("kernel_edgedetect_5x5", MPIX_OP_KERNEL_CONVOLVE_5X5,
+				    MPIX_FMT_RGB24, MPIX_KERNEL_EDGE_DETECT),
+		BENCHMARK_DESC_AUTO("kernel_gaussianblur_3x3", MPIX_OP_KERNEL_CONVOLVE_3X3,
+				    MPIX_FMT_RGB24, MPIX_KERNEL_GAUSSIAN_BLUR),
+		BENCHMARK_DESC_AUTO("kernel_gaussianblur_5x5", MPIX_OP_KERNEL_CONVOLVE_5X5,
+				    MPIX_FMT_RGB24, MPIX_KERNEL_GAUSSIAN_BLUR),
+		BENCHMARK_DESC_AUTO("kernel_denoise_3x3", MPIX_OP_KERNEL_DENOISE_3X3,
+				    MPIX_FMT_RGB24),
+		BENCHMARK_DESC_AUTO("kernel_denoise_5x5", MPIX_OP_KERNEL_DENOISE_5X5,
+				    MPIX_FMT_RGB24),
+
+		/* resize */
+		BENCHMARK_DESC_AUTO("resize_subsample", MPIX_OP_RESIZE_SUBSAMPLE, MPIX_FMT_RGB24,
+				    WIDTH, HEIGHT),
+
+		/* image encoding */
+		BENCHMARK_DESC_AUTO("jpeg_encode", MPIX_OP_JPEG_ENCODE, MPIX_FMT_RGB24),
+		BENCHMARK_DESC_AUTO("qoi_encode", MPIX_OP_QOI_ENCODE, MPIX_FMT_RGB24),
+
+		/* palettize */
+		// {MPIX_FMT_RGB24, benchmark_run_palettize},
```

Comment:
this one is a special case as it requires extra function call to set the palette. can be kept as a comment for now like you did.